src/main/java/de/hysky/skyblocker/skyblock/tabhud/config/preview/PreviewWidget.java

```diff
@@ -0,0 +1,303 @@
+package de.hysky.skyblocker.skyblock.tabhud.config.preview;
+
+import de.hysky.skyblocker.config.SkyblockerConfigManager;
+import de.hysky.skyblocker.mixins.accessors.InGameHudInvoker;
+import de.hysky.skyblocker.skyblock.tabhud.screenbuilder.ScreenBuilder;
+import de.hysky.skyblocker.skyblock.tabhud.screenbuilder.ScreenMaster;
+import de.hysky.skyblocker.skyblock.tabhud.screenbuilder.pipeline.PositionRule;
+import de.hysky.skyblocker.skyblock.tabhud.widget.HudWidget;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.DrawContext;
+import net.minecraft.client.gui.ScreenPos;
+import net.minecraft.client.gui.screen.narration.NarrationMessageBuilder;
+import net.minecraft.client.gui.widget.ClickableWidget;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.text.Text;
+import net.minecraft.util.Colors;
+import net.minecraft.util.Formatting;
+import org.jetbrains.annotations.Nullable;
+import org.lwjgl.glfw.GLFW;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * The preview widget that captures clicks and displays the current state of the widgets.
+ */
+public class PreviewWidget extends ClickableWidget {
+
+	private final PreviewTab tab;
+	float ratio = 1f;
```

Comment:
is it possible to use a more narrow access modifier (couldn't check the usages of this so maybe I am wrong)

Reply:
used by PreviewTab